java-components/management-console/src/main/java/com/redhat/hacbs/management/resources/BuildLogs.java

```diff
@@ -0,0 +1,111 @@
+package com.redhat.hacbs.management.resources;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.util.LinkedHashSet;
+import java.util.Map;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import com.redhat.hacbs.common.tools.completer.BuildCompleter;
+import com.redhat.hacbs.common.tools.logging.LogExtractor;
+import com.redhat.hacbs.resources.model.v1alpha1.DependencyBuild;
+
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.openshift.api.model.Route;
+import io.fabric8.openshift.api.model.RouteSpec;
+import io.fabric8.openshift.client.OpenShiftClient;
+import io.fabric8.tekton.pipeline.v1beta1.PipelineRun;
+import io.quarkus.arc.Arc;
+import io.quarkus.logging.Log;
+import software.amazon.awssdk.services.s3.S3Client;
+
+public class BuildLogs {
+
+    protected enum Type {
+        DISCOVERY,
+        BUILD
+    }
+
+    private static final String DEV_PATH = "/apis/results.tekton.dev";
+
+    private static final String PROD_PATH = "/api/k8s/plugins/tekton-results/workspaces/";
+
+    @Inject
+    S3Client s3Client;
+
+    @ConfigProperty(name = "JBS_S3_SYNC_ENABLED", defaultValue = "true")
```

Comment:
This should be jbs.s3.sync.enabled, the config system automatically maps upper case env vars.

Reply:
Also this will mean that by default the CLI will only pull logs from S3 unless you explicitly set this, it would be better if we could be smarter about this.

Reply:
i.e. try S3 by default, then fall back to the other methods if it fails. Only print out the exception details if none of them work.

Reply:
I think its the inverse - it will always attempt to try to pull logs from S3 as it defaults to true? 

However, I do agree that it doesn't do a fallback which the CLI does (partially).